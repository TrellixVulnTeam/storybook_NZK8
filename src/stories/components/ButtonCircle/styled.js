// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

/**
 * Determine the button's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.brightBlue;
    case 'red':
      return colors.strawberry;
    default:
      return '';
  }
}

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 34px;
  height: 34px;
  line-height: 34px;
  background: ${p => getAccentColor(p.theme)};

  border-radius: 50%;
  color: ${colors.white};
  cursor: pointer;

  span {
    width: 100%;
    text-align: center;
  }

  &:hover {
    box-shadow: 0 2px 10px 0 rgba(40, 44, 49, 0.2);
  }
`;
