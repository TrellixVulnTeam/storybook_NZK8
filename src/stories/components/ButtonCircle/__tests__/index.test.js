import React from 'react';
import { mount } from 'enzyme';
import ButtonCircle from 'components/ButtonCircle/index';

describe('<ButtonCircle />', () => {
  it('should render correctly without theme', () => {
    const sut = mount(<ButtonCircle />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with red theme', () => {
    const sut = mount(<ButtonCircle theme="red" />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with blue theme', () => {
    const sut = mount(<ButtonCircle theme="blue" />);

    expect(sut).toMatchSnapshot();
  });
});
