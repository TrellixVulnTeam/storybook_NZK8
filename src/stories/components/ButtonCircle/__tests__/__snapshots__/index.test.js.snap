// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCircle /> should render correctly with blue theme 1`] = `
<_default
  theme="blue"
>
  <styled__Wrapper
    theme="blue"
  >
    <button
      className="styled__Wrapper-akfokf-0 hQjSbH"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<ButtonCircle /> should render correctly with red theme 1`] = `
<_default
  theme="red"
>
  <styled__Wrapper
    theme="red"
  >
    <button
      className="styled__Wrapper-akfokf-0 hgfigR"
    />
  </styled__Wrapper>
</_default>
`;

exports[`<ButtonCircle /> should render correctly without theme 1`] = `
<_default>
  <styled__Wrapper>
    <button
      className="styled__Wrapper-akfokf-0 faeiwv"
    />
  </styled__Wrapper>
</_default>
`;
